rebasehelper/plugins/spec_hooks/replace_old_version.py

```diff
@@ -114,18 +98,33 @@ def run(cls, spec_file: SpecFile, rebase_spec_file: SpecFile, **kwargs: Any):
         replace_with_macro = bool(kwargs.get('replace_old_version_with_macro'))
 
         subversion_patterns = cls._create_possible_replacements(old_version, new_version, replace_with_macro)
+        examined_lines = set()
+        for tag in rebase_spec_file.tags.filter():
+            examined_lines.add(tag.line)
+            value = rebase_spec_file.spec_content.section(tag.section)[tag.line][slice(*tag.value_span)]
+            if tag.name in cls.IGNORED_TAGS:
+                continue
+            scheme = urllib.parse.urlparse(value).scheme
+            if (tag.name.startswith('Patch') or tag.name.startswith('Source')) and not scheme:
+                # skip local sources
+                continue
+
+            # replace the whole version first
+            updated_line = subversion_patterns[0][0].sub(subversion_patterns[0][1], value)
+            # replace subversions only for remote sources/patches
+            if (tag.name.startswith('Patch') or tag.name.startswith('Source')) and urllib.parse.urlparse(updated_line):
+                for sub_pattern, repl in subversion_patterns[1:]:
+                    updated_line = sub_pattern.sub(repl, updated_line)
+            rebase_spec_file.set_raw_tag_value(tag.name, updated_line, tag.section)
+
         for sec_name, section in rebase_spec_file.spec_content.sections:
             if sec_name.startswith('%changelog'):
                 continue
             for index, line in enumerate(section):
-                if cls._is_local_source(line) or any(line.startswith(tag) for tag in cls.IGNORED_TAGS):
+                if index in examined_lines:
```

Comment:
How is this supposed to work? `examined_lines` contains line numbers from preamble, but you are not checking section here at all.